### DISCUSSION TITLE
Cambio de orden pinguinos

### DISCUSSION THREAD

**Discussion by Lexiesm** [IIC2133-PUC/2024-2#72] [Tarea 0] [answered]
Hola! tengo un problema de orden, con valgrind no me tira errores y ya no se cómo solucionarlo :(. Sucede que por ejemplo en el test 1 hard me arroja todo bien excepto el orden de dos pinguinos:
![imagen](https://github.com/user-attachments/assets/9c8f63bc-ab91-4208-aa85-105b47272c7a)

pero el valdgrind no tira errores:
![imagen](https://github.com/user-attachments/assets/0d9e893a-f0c6-4288-8bee-9fb8509c416d)

lo mismo sucede con test2 easy2, med1, med2, hard1 y hard2 :( valgrind no me tira errores y para los test 1 easy, med y test 2 easy 1 no sucede. He intentado ver la lógica de mi código pero no veo algo que me diga por qué pasando, de hecho en el test 1 hard, pasa cuando llega el pinguino 11, 20 y 15 (luego se va el 11) pero me lo ordena como 15 y 20 en vez de 20 y 15, en otras partes también ocurren secuencias como esas pero no sucede ese cambio. 
No se si tendrán una idea de qué podría ser :(

de antemano gracias

**Comment by tomastrivino** [comment, marked as answer]
Hola,
esto podría tener algo que ver con la forma en que estás modelando el almacenamiento de los pingüinos. ¿Cuál es tu criterio para hacer el print? ¿Estás printeando por id, por orden de ingreso, por posición relativa? etc etc. (Son preguntas que no busco que me respondas acá, sino que las pienses bien). Repasa el enunciado y presta especial atención a cosas como el orden en que se deben printear. Además, intenta con tests lo más chicos posibles primero para detectar bien la fuente del error.

No te preocupes por ahora por lo de Valgrind, tu programa no tiene errores de memoria, por lo que el problema acá es a nivel de la lógica que estás implementando en tu código en algún lado.

Saludos

**Reply by Lexiesm** [reply]
efectivamente, tenía un tema al "sacar" los pingüinos cuando se iban y reincorporar los nuevos que no estaba considerando, muchas gracias :)
